### PR TITLE
Test for binary grep output in cram tests

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
PR to supply test for #5417.

### CODE REVIEW

File: test/blackbox-tests/test-cases/cram/grep-z.t [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Grep should work
+
+  $ cat > multi-line.opam <<EOF
+  > x-before: "yes"
+  > depends: [
+  >   "a"
+  > ]
+  x-after: "true"
+  EOF
```
Comment (emillon) [inline test/blackbox-tests/test-cases/cram/grep-z.t:11]:
```suggestion
  > x-after: "true"
  > EOF
```

**Comment by emillon** [comment]
It's not that much that `grep` fails, it's that `git diff` thinks that the corrected file is binary.
`grep` is doing something with its output because piping to `xxd` reveals no NUL bytes, and piping to `cat` displays the output normally.

**Comment by emillon** [comment]
Grep adds a nul byte at the end but only when it's not connected to a pipe:
```
+  $ grep -Pzo "(?s)depends:.*?]\n"< multi-line.opam>out
 
   $ xxd out
+  00000000: 6465 7065 6e64 733a 205b 0a20 2022 6122  depends: [.  "a"
+  00000010: 0a5d 0a00                                .]..
```

**Comment by Leonidas-from-XIV** [comment]
An option would be to force dune to call `git grep` with the `-a`/`--text` option, since the `.corrected` file should be text, just as the input file needs to be. That would at least show the difference of what is going on.

**Comment by emillon** [comment]
If your test displays binary data, I think it's fair for dune not to display the corrected file in that case. I'd say that the correct fix is not to output the nul byte in the cram test.

**Comment by Leonidas-from-XIV** [comment]
I sort of agree in principle, but my file was binary by accident and given cram tests are assumed to be both text in and text out I think it would be good if dune did a better job at reporting the issue. In this case I couldn't even figure out where I accidentally generated the NUL byte and had to bisect and that's not a good user experience.

**Pull request closed without merging**